### PR TITLE
[s]Stops the create area search early if too big

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
This proc is very expensive when you run it in a very big area. There is no need for feedback past two times as big as allowed, and it stops potential abuse.

## Why It's Good For The Game
Makes servers harder to break.

## Changelog
**cobby edit**: not needed

### CODE REVIEW

File: code/__HELPERS/areas.dm [hunk 1]
```diff
@@ -53,7 +55,7 @@ GLOBAL_LIST_INIT(typecache_powerfailure_safe_areas, typecacheof(/area/engine/eng
 		to_chat(creator, "<span class='warning'>The new area must be completely airtight and not a part of a shuttle.</span>")
 		return
 	if(turfs.len > BP_MAX_ROOM_SIZE)
-		to_chat(creator, "<span class='warning'>The room you're in is too big. It is [((turfs.len / BP_MAX_ROOM_SIZE)-1)*100]% larger than allowed.</span>")
+		to_chat(creator, "<span class='warning'>The room you're in is too big. It is [turfs.len >= BP_MAX_ROOM_SIZE *2 ? "more than 200" : ((turfs.len / BP_MAX_ROOM_SIZE)-1)*100]% larger than allowed.</span>")
```
Comment (SpaceManiac) [inline code/__HELPERS/areas.dm]: Good PR, little off in the math - 2x as large is 100% larger, not 200% larger

Comment (kriskog) [reply code/__HELPERS/areas.dm]: This is true. I'll fix that haha

**Review by SpaceManiac — APPROVED** [review]

**Comment by fludd12** [comment]
Changelog isn't necessary, as users won't see the effect of the change - and it will keep the issue [S]ilent to any servers that haven't updated as quickly that borrow from our codebase.

**Comment by kriskog** [comment]
Fair enough

File: code/__HELPERS/areas.dm [hunk 2]
```diff
@@ -22,6 +22,8 @@ GLOBAL_LIST_INIT(typecache_powerfailure_safe_areas, typecacheof(/area/engine/eng
 		found_turfs.Cut(1, 2)
 		var/dir_flags = checked_turfs[sourceT]
 		for(var/dir in GLOB.alldirs)
+			if(length(.) > 2 * BP_MAX_ROOM_SIZE)
+				return
```
Comment (ninjanomnom) [inline code/__HELPERS/areas.dm]: This proc isnt just for blueprints, like I said last time this was suggested

Comment (kriskog) [reply code/__HELPERS/areas.dm]: detect_room is only used by create_area as far as I can see though, so nothing else should be affected.

Comment (ninjanomnom) [reply code/__HELPERS/areas.dm]: You misunderstand, the proc itself has a simple purpose independent of the blueprints. This is useful for things other than it and one of the reasons I originaly wrote it was to seperate it out for some other features people wanted. 

Those didn't really turn out but the purpose of the proc is still there. Inserting snowflake exceptions like this into unrelated code is a perfect way to cultivate shitcode.

Comment (kriskog) [reply code/__HELPERS/areas.dm]: Fair enough, I'll make a different fix for it. We do need the proc to stop early when called by create_area though, but I can add an argument that defaults to ignore size. Sound good to you?

**Pull request merged**